p2p/peer.go

```diff
@@ -70,7 +179,7 @@ type PeerUpdatesCh struct {
 // NewPeerUpdates returns a reference to a new PeerUpdatesCh.
 func NewPeerUpdates() *PeerUpdatesCh {
 	return &PeerUpdatesCh{
-		updatesCh: make(chan PeerUpdate),
+		updatesCh: make(chan PeerUpdate, 1),
```

Comment:
Curious, what was the motivation to make this buffered? Would this ever need to be configurable?

Reply:
Yeah, so this may be a bit of premature optimization. In the router, whenever there's a peer change it needs to be broadcast across all peer updates subcriptions by looping over them. If this isn't buffered, the scheduler needs to do a context switch for each iteration of the loop (it blocks until the goroutine on the other side gets scheduled and receives the update). A buffer of 1 allows the for-loop to complete without any context switches, while still making sure all subscribers are reasonably in sync (since there's only 1, they'll all be processing the same peer update at roughly the same time).

This should really be benchmarked, I added it purely on intuition. Would be happy to replace it with a `FIXME` to investigate later.

Reply:
This makes sense. I think we can and probably should add a `FIXME` though.